anydex/wallet/iota/iota_wallet.py

```diff
@@ -0,0 +1,334 @@
+import os
+import re
+from abc import ABCMeta
+from asyncio import Future
+
+from iota.crypto.types import Seed
+from iota.transaction import ProposedTransaction
+from iota.types import Address
+from ipv8.util import succeed, fail
+from sqlalchemy import exists
+
+from anydex.wallet.cryptocurrency import Cryptocurrency
+from anydex.wallet.iota.iota_database import initialize_db, \
+    DatabaseSeed, DatabaseTransaction, DatabaseBundle, DatabaseAddress
+from anydex.wallet.iota.iota_provider import IotaProvider
+from anydex.wallet.wallet import Wallet, InsufficientFunds
+
+
+class AbstractIotaWallet(Wallet, metaclass=ABCMeta):
+
+    def __init__(self, db_path, testnet, node):
```

Comment:
In case the `node` parameter is meant to be of type Node, I think it's useful to use a type hint there.
Also because in the provider class, a `node` is a string.

Reply:
Same holds for the `testnet` parameter.

Reply:
Node is meant to be a string and testnet a boolean. I can update it to require a specific type.

Reply:
Added type enforcement 👍